src/app/workflow/executions/executions-tab.component.ts

```diff
@@ -0,0 +1,218 @@
+import PartnerEnum = CloudInstance.PartnerEnum;
+import { MatSelectChange } from '@angular/material/select';
+import { AlertService } from '../../shared/alert/state/alert.service';
+
+interface MetricsTableObject {
+  metric: string;
+  min: number;
+  avg: number;
+  max: number;
+  unit: string;
+}
+interface ValidationsTableObject {
+  validatorTool: string;
+  mostRecentVersion: string;
+  isValid: boolean;
+  mostRecentErrorMessage: string;
+  successfulValidationVersions: string[];
+  failedValidationVersions: string[];
+  numberOfRuns: number;
+  passingRate: number;
+}
+
+@Component({
+  selector: 'app-executions-tab',
+  templateUrl: './executions-tab.component.html',
+  styleUrls: ['./executions-tab.component.css'],
+})
+export class ExecutionsTabComponent extends EntryTab implements OnChanges {
+  metrics: Map<PartnerEnum, Metrics>;
+  trsID: string;
+  currentPartner: PartnerEnum;
+  partners: PartnerEnum[];
+  metricsTable: MetricsTableObject[];
+  validationsTable: ValidationsTableObject[];
+  metricsColumns: string[];
+  validationsColumns: string[];
+  totalExecutions: number;
+  successfulExecutions: number;
+  failedExecutions: number;
+  metricsExist: boolean;
+
+  @Input() entry: BioWorkflow | Service | Notebook;
+  @Input() version: WorkflowVersion;
+
+  constructor(
+    private extendedGA4GHService: ExtendedGA4GHService,
+    private workflowQuery: WorkflowQuery,
+    private alertService: AlertService,
+    protected sessionQuery: SessionQuery,
+    private checkerWorkflowService: CheckerWorkflowService,
+    private checkerWorkflowQuery: CheckerWorkflowQuery
+  ) {
+    super();
+  }
+
+  ngOnChanges() {
+    this.alertService.start('Retrieving metrics data');
+    this.trsID = this.checkerWorkflowQuery.getTRSId(this.entry);
+    this.extendedGA4GHService
+      .aggregatedMetricsGet(this.trsID, this.version.name)
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe(
+        (metrics) => {
+          this.resetMetricsData();
+          if (metrics) {
+            for (const [partner, metric] of Object.entries(metrics)) {
+              this.setMetricsObject(this.metrics, partner, metric);
+            }
+            this.partners = Array.from(this.metrics.keys());
+            if (this.partners.length > 0) {
+              this.metricsExist = true;
+              this.selectPartner(this.partners[0]);
+            }
+          }
+          this.alertService.detailedSuccess();
+        },
+        (error) => {
+          this.alertService.detailedError(error);
+        }
+      );
+  }
+
+  private setMetricsObject(metricsMap: Map<PartnerEnum, Metrics>, partner: string, metric: Metrics) {
+    for (const partnerEnum in PartnerEnum) {
+      if (PartnerEnum[partnerEnum] === partner) {
+        metricsMap.set(PartnerEnum[partnerEnum], metric);
+      }
+    }
+  }
+
+  private resetMetricsData() {
+    this.metricsExist = false;
+    this.metrics = new Map();
+    this.metricsTable = null;
```

Comment:
We're wildly inconsistent on this, but you should probably set this and validationsTable to an empty array instead of null. Or change the definition on line 65 to `metricsTable: MetricsTableObject[] | null`.